### PR TITLE
Reduce signatures - drop redundant BLS in QBFT (no fork)

### PR DESCRIPTION
**Pull request by MatheusFranco99**
# Overview

This PR drops redundant BLS signature checks in QBFT.

## Changes
- [x] Drop redundant checks for Proposal, Prepare, Commit and Round-Change messages
- [x] Keep checks for message justification and decided messages
- [x] Add a check to ensure that the signers belong to the committee

### CODE REVIEW

File: qbft/prepare.go [hunk 1]
```diff
@@ -108,8 +111,14 @@ func validSignedPrepareForHeightRoundAndRoot(
 		return errors.New("msg allows 1 signer")
 	}
 
-	if err := signedPrepare.Signature.VerifyByOperators(signedPrepare, config.GetSignatureDomainType(), types.QBFTSignatureType, operators); err != nil {
-		return errors.Wrap(err, "msg signature invalid")
+	if !signedPrepare.CheckSignersInCommittee(operators) {
+		return errors.New("signers not in committee")
+	}
```
Comment (GalRogozinski) [inline qbft/prepare.go:116]: just from sw engineering point of view
shouldn't we have some `baseValidation` that keeps on calling this

so we never forget to add this?
because it repeats itself

**Review by GalRogozinski — APPROVED** [review]

**Review by GalRogozinski — APPROVED** [review]
Approved but you can still fix my comment

File: qbft/round_change.go [hunk 2]
```diff
@@ -213,6 +214,7 @@ func validRoundChangeForData(
 	height Height,
 	round Round,
 	fullData []byte,
+	verifySignature bool,
```
Comment (moshe-blox) [inline qbft/round_change.go:217]: switch to dontVerifySignature?

Comment (moshe-blox) [reply qbft/round_change.go:217]: on another thought, maybe we should go for the validateX() and validateXNoVerify() pattern?

File: qbft/commit.go [hunk 3]
```diff
@@ -127,7 +135,7 @@ func validateCommit(
 	proposedMsg *SignedMessage,
 	operators []*types.Operator,
 ) error {
-	if err := baseCommitValidation(config, signedCommit, height, operators); err != nil {
+	if err := baseCommitValidation(config, signedCommit, height, operators, false); err != nil {
```
Comment (moshe-blox) [inline qbft/commit.go:138]: if we go for the NoVerify suffix pattern, validateCommit could be validateCommitNoVerify to be explicit

**Review by moshe-blox — APPROVED** [review]

**Review by moshe-blox — CHANGES_REQUESTED** [review]
Let's merge this after #345 is on main

File: qbft/round_change.go (hunk 2 above)
Comment (MatheusFranco99) [reply qbft/round_change.go:217]: Agree

File: qbft/commit.go (hunk 3 above)
Comment (MatheusFranco99) [reply qbft/commit.go:138]: Agree

File: qbft/commit.go [hunk 4]
```diff
@@ -111,9 +113,15 @@ func baseCommitValidation(
 		return errors.Wrap(err, "signed commit invalid")
 	}
 
+	if !signedCommit.CheckSignersInCommittee(operators) {
+		return errors.New("signers not in committee")
+	}
```
[the remark this answers is not in the document]
Comment (MatheusFranco99) [reply qbft/commit.go:118]: I think it does. But I'm leaving it here because spec doesn't know and it's a light check

File: qbft/prepare.go (hunk 1 above)
Comment (MatheusFranco99) [reply qbft/prepare.go:114]: I think we should totally do that. Not only for this check but for many others such as height, msg.Validate(), etc. Since it would require some bigger refactoring, let's do it in a future PR? Just so we can dispatch this to implementation as soon as possible and as simple as possible.

File: qbft/commit.go [hunk 5]
```diff
@@ -94,7 +95,7 @@ func CreateCommit(state *State, config IConfig, root [32]byte) (*SignedMessage,
 	return signedMsg, nil
 }
 
-func baseCommitValidation(
+func baseCommitValidationNoVerification(
```
Comment (olegshmuelov) [inline qbft/commit.go:98]: the config is not used here anymore, should we delete it?

File: qbft/prepare.go [hunk 6]
```diff
@@ -79,7 +80,7 @@ func getRoundChangeJustification(state *State, config IConfig, prepareMsgContain
 
 // validSignedPrepareForHeightRoundAndRoot known in dafny spec as validSignedPrepareForHeightRoundAndDigest
 // https://entethalliance.github.io/client-spec/qbft_spec.html#dfn-qbftspecification
-func validSignedPrepareForHeightRoundAndRoot(
+func validSignedPrepareForHeightRoundAndRootNoVerification(
```
Comment (olegshmuelov) [inline qbft/prepare.go:83]: the config is not used here anymore, should we delete it?

Comment (MatheusFranco99) [reply qbft/prepare.go:83]: Indeed. I'll remove it. Thanks

File: qbft/commit.go (hunk 5 above)
Comment (MatheusFranco99) [reply qbft/commit.go:98]: Indeed. I'll remove it. Thanks

File: qbft/round_change.go [hunk 7]
```diff
@@ -227,12 +228,12 @@ func validRoundChangeForData(
 		return errors.New("msg allows 1 signer")
 	}
 
-	if err := signedMsg.Signature.VerifyByOperators(signedMsg, config.GetSignatureDomainType(), types.QBFTSignatureType, state.Share.Committee); err != nil {
-		return errors.Wrap(err, "msg signature invalid")
+	if err := signedMsg.Validate(); err != nil {
+		return errors.Wrap(err, "roundChange invalid")
```
Comment (olegshmuelov) [inline qbft/round_change.go:232]: just to understand, it was a bug that we validated only the inner message?

Comment (MatheusFranco99) [reply qbft/round_change.go:232]: I think so

Comment (olegshmuelov) [reply qbft/round_change.go:232]: if so, we have to add a test for `message signers is empty`, `non unique signer` & `signer ID 0 not allowed`

Comment (MatheusFranco99) [reply qbft/round_change.go:232]: Yup, adding these tests.
Btw, the previous version couldn't produce any errors because, even though it should be `signedMsg.Validate()` and not `signedMsg.Message.Validate()`, the other checks were also verifying the conditions. But, of course, still better to call `signedMsg.Validate()`

**Review by olegshmuelov — APPROVED** [review]

File: qbft/commit.go [hunk 8]
```diff
@@ -127,7 +145,7 @@ func validateCommit(
 	proposedMsg *SignedMessage,
 	operators []*types.Operator,
 ) error {
-	if err := baseCommitValidation(config, signedCommit, height, operators); err != nil {
+	if err := baseCommitValidationNoVerification(signedCommit, height, operators); err != nil {
```
Comment (olegshmuelov) [inline qbft/commit.go:148]: one more cleanup can be done here
to not pass the `config` to validateCommit

**Comment by olegshmuelov** [comment]
since https://github.com/bloxapp/ssv-spec/pull/342 was merged already


can we fix small [issue](https://github.com/bloxapp/ssv-spec/pull/342#discussion_r1542515196) in the current pr?

**Comment by GalRogozinski** [comment]
> since #342 was merged already
> 
> can we fix small [issue](https://github.com/bloxapp/ssv-spec/pull/342#discussion_r1542515196) in the current pr?

This PR doesn't change runner at all...
Maybe in a subsequent PR when we change runner

**Comment by olegshmuelov** [comment]
> > since #342 was merged already
> > can we fix small [issue](https://github.com/bloxapp/ssv-spec/pull/342#discussion_r1542515196) in the current pr?
> 
> This PR doesn't change runner at all... Maybe in a subsequent PR when we change runner

created an issue
https://github.com/bloxapp/ssv-spec/issues/376

File: types/operator.go [hunk 9]
```diff
@@ -5,13 +5,19 @@ type OperatorID = uint64
 
 // Operator represents an SSV operator node
 type Operator struct {
-	OperatorID OperatorID
-	PubKey     []byte `ssz-size:"48"`
+	OperatorID    OperatorID
+	BeaconPubKey  []byte `ssz-size:"48"`
```
Comment (olegshmuelov) [inline types/operator.go:9]: Can we consider calling it SharePubKey?

Comment (MatheusFranco99) [reply types/operator.go:9]: Yes, better 👍

File: qbft/commit.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (y0sher) [reply qbft/commit.go:116]: I think we should NEVER assume something happens in msg validation hence we don't need to take care in consensus.

**Review by GalRogozinski — CHANGES_REQUESTED** [review]
I don't see new sig verification tests in `runner/consensus` pkg

File: qbft/commit.go (hunk 5 above)
Comment (MatheusFranco99) [reply qbft/commit.go:98]: Yeah. "IgnoreSignature" is more clear. I'll update it. Thanks!

File: qbft/round_change.go (hunk 7 above)
Comment (MatheusFranco99) [reply qbft/round_change.go:232]: It is not a bug, but all other base validation functions for the other message types call validate on the upper message, and this is the only inconsistent one. Even though, all of these calls are indeed duplicated 😅

File: types/signer.go [hunk 10]
```diff
@@ -35,6 +37,7 @@ type BeaconSigner interface {
 // SSVSigner used for all SSV specific signing
 type SSVSigner interface {
 	SignRoot(data Root, sigType SignatureType, pk []byte) (Signature, error)
+	SignNetworkData(data []byte, pk []byte) ([]byte, error)
```
[the remark this answers is not in the document]
Comment (MatheusFranco99) [reply types/signer.go:40]: I find `OperatorData` confusing

File: types/testingutils/keymanager.go [hunk 11]
```diff
@@ -22,6 +25,7 @@ type testingKeyManager struct {
 	keys           map[string]*bls.SecretKey
 	ecdsaKeys      map[string]*ecdsa.PrivateKey
 	encryptionKeys map[string]*rsa.PrivateKey
+	networkKeys    map[string]*rsa.PrivateKey
```
[the remark this answers is not in the document]
Comment (MatheusFranco99) [reply types/testingutils/keymanager.go:28]: I don't know what these `encryptionKeys` are for. It's never used in spec. Maybe some old deprecated functionality once used it and now it's a leftover.

I find the name `OperatorKey` confusing. The operator has many key types. I think we should include its purpose in the name, e.g. network, share, beacon, etc.

File: types/testingutils/keymanager.go [hunk 12]
```diff
@@ -166,11 +154,30 @@ func (km *testingKeyManager) SignETHDepositRoot(root []byte, address common.Addr
 	panic("implement")
 }
 
+func (km *testingKeyManager) SignNetworkData(data []byte, pk []byte) ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (MatheusFranco99) [reply types/testingutils/keymanager.go:157]: See above comment

File: types/testingutils/keymanager.go [hunk 13]
```diff
@@ -166,11 +154,30 @@ func (km *testingKeyManager) SignETHDepositRoot(root []byte, address common.Addr
...
+	}
+	return signature, nil
+}
+
 func (km *testingKeyManager) AddShare(shareKey *bls.SecretKey) error {
 	km.keys[hex.EncodeToString(shareKey.GetPublicKey().Serialize())] = shareKey
 	return nil
 }
 
+func (km *testingKeyManager) AddNetworkKey(sk *rsa.PrivateKey) error {
```
[the remark this answers is not in the document]
Comment (MatheusFranco99) [reply types/testingutils/keymanager.go:172]: See above comment

File: qbft/spectest/tests/commit/unknwon_signer.go [hunk 14]
```diff
@@ -23,9 +23,9 @@ func UnknownSigner() tests.SpecTest {
 	return &tests.MsgProcessingSpecTest{
 		Name:          "unknown commit signer",
 		Pre:           pre,
-		PostRoot:      "976cd5cecd58bba892a38ec0ef02b3aed4656fb89fef473d8af78fedf095439d",
+		PostRoot:      "f92657c2f9b7225573bcd5b03a0f59dfda3f04c361e022e37357bf22c7a10ab9",
 		InputMessages: msgs,
-		ExpectedError: "invalid signed message: msg signature invalid: unknown signer",
+		ExpectedError: "invalid signed message: signers not in committee",
```
[the remark this answers is not in the document]
Comment (MatheusFranco99) [reply qbft/spectest/tests/commit/unknwon_signer.go:28]: Indeed. I'm going to update it. Thanks!

File: types/testingutils/keymanager.go (hunk 11 above)
[the remark this answers is not in the document]
Comment (GalRogozinski) [reply types/testingutils/keymanager.go:28]: The encryption key is what is used to encrypt on chain validator keys.
It is the same key that we use for signing message.

Why is this part of spec? Well basically it depends on what is the scope of spec... and lately I've come to realize that different people at the company expect different things from spec... We wanted to remove it in #348... But @lior-blox didn't answer my question yet :-\

Conceptually there could be two different keys (for encryption and signing), and one can claim the implementation decides to use the same key for both operations...
So rethinking this, both having 2 keys or one key can pass my review, I won't fuss :-)

But I don't agree an operator has many key types (mostly 2)... The `shareKey` is linked to a validator.
Sure an operators hold validator so I can see why is this confusing.

But I really dislike `networkKey` especially that now in spec we decided to move authentication to `Validator`...
Maybe a name we can agree on is `SSVKey`?
It matches the `SSVSigner` interface. Also it signifies those keys are used specifically by the SSV protocol (unlike the SharedKeys that are Eth specific)

File: types/signer.go (hunk 10 above)
[the remark this answers is not in the document]
Comment (GalRogozinski) [reply types/signer.go:40]: Maybe `SSVData`...
I don't know I find `NetworkData` confusing as long as we don't put it in p2p package.
See https://github.com/bloxapp/ssv-spec/pull/372#discussion_r1550196788

File: types/operator.go [hunk 15]
```diff
@@ -5,13 +5,19 @@ type OperatorID = uint64
...
+	NetworkPubKey []byte `ssz-size:"294"`
 }
 
-// GetPublicKey returns the public key with which the node is identified with
-func (n *Operator) GetPublicKey() []byte {
-	return n.PubKey
+// GetSharePublicKey returns the share public key
+func (n *Operator) GetSharePublicKey() []byte {
+	return n.SharePubKey
+}
```
[the remark this answers is not in the document]
Comment (GalRogozinski) [reply types/operator.go:16]: > But I think this should be done in another PR.
Aite, too many diffs are bad at this point 👍 

> Even though spec may not care about memory, since the implementation is forced to be aligned to spec, I think we should care and implement it in the future PR.
Not important for me, but we can. If @moshe-blox will feel this is important then we will do it for sure.

File: types/signer.go [hunk 16]
```diff
@@ -53,4 +56,6 @@ type KeyManager interface {
 	AddShare(shareKey *bls.SecretKey) error
 	// RemoveShare removes a share key
 	RemoveShare(pubKey string) error
+	// AddNetworkKey saves a network key
+	AddNetworkKey(sk *rsa.PrivateKey) error
```
Comment (GalRogozinski) [inline types/signer.go:60]: So after discussing with @y0sher and @olegshmuelov

Get rid of the keyManager interface. `Share` functions should move to `BeaconSigner`. `Network` (rename to SSV) functions should go to SSVSigner

The `KeyManager` in testingutils can implement both to reduce changes (but you can do some other design if you wish)

**Review by GalRogozinski — CHANGES_REQUESTED** [review]

**Review by GalRogozinski — CHANGES_REQUESTED** [review]
Another point is that our Broadcasted messages should be signed.

I think that if you change `testingutils.Broadcast` it is good enough since it will change all the tests. It will also be like impl that currently put in p2p

File: types/testingutils/keymanager.go (hunk 11 above)
[the remark this answers is not in the document]
Comment (MatheusFranco99) [reply types/testingutils/keymanager.go:28]: The same key is being used for both encryption and signature?

Wait, but the `SSVSigner` according to [comment](https://github.com/bloxapp/ssv-spec/pull/372#discussion_r1550144782) is already a name being used for the BLS shares sign.

**Comment by GalRogozinski** [comment]
Also we need to maybe add an `RSAVerify` to MsgProcessing test on the broadcasted message to ensure that messages were signed correctly

**Comment by GalRogozinski** [comment]
Also we need to drop the double verification at the p2p layer now...

File: ssv/attester.go [hunk 17]
```diff
@@ -227,6 +230,10 @@ func (r *AttesterRunner) GetSigner() types.KeyManager {
 	return r.signer
 }
 
+func (r *AttesterRunner) GetOperatorSigner() types.SSVOperatorSigner {
+	return r.operatorSigner
+}
+
```
Comment (GalRogozinski) [inline ssv/attester.go:236]: does runner use the signer?
in the end we use the network to sign it

Comment (MatheusFranco99) [reply ssv/attester.go:236]: Indeed. Removing it. Thanks!

File: qbft/state.go [hunk 18]
```diff
@@ -29,20 +31,26 @@ type IConfig interface {
...
+	Domain         types.DomainType
+	ValueCheckF    ProposedValueCheckF
+	ProposerF      ProposerF
+	Network        Network
+	Timer          Timer
 }
 
 // GetSigner returns a Signer instance
-func (c *Config) GetSigner() types.SSVSigner {
+func (c *Config) GetSigner() types.SSVShareSigner {
```
[the remark this answers is not in the document]
Comment (MatheusFranco99) [reply qbft/state.go:45]: Yes, better. Thanks

File: types/testingutils/operator_signer.go [hunk 19]
```diff
@@ -16,10 +17,14 @@ type testingOperatorSigner struct {
 
 var (
 	testingOperatorSignerInstance *testingOperatorSigner
+	opMu                          sync.Mutex
```
[the remark this answers is not in the document]
Comment (MatheusFranco99) [reply types/testingutils/operator_signer.go:20]: The lack of it randomly produces errors around tests

File: qbft/state.go [hunk 20]
```diff
@@ -31,7 +31,7 @@ type IConfig interface {
 }
 
 type Config struct {
-	Signer         types.SSVShareSigner
+	SSVShareSigner types.SSVShareSigner
```
Comment (moshe-blox) [inline qbft/state.go:34]: this should probably be just `ShareSigner` like below, and the same for method names

tbh im not sure that we even need the word 'SSV' in the structs, but it's not too bad, lmk what you think @GalRogozinski @MatheusFranco99

**Review by moshe-blox — CHANGES_REQUESTED** [review]

File: qbft/state.go (hunk 20 above)
Comment (MatheusFranco99) [reply qbft/state.go:34]: I agree the word `SSV` here is redundant. I'll remove it, thx

File: qbft/round_change.go [hunk 21]
```diff
@@ -271,6 +272,28 @@ func validRoundChangeForData(
 
 		return nil
 	}
+
+	return nil
+}
+
+func validRoundChangeForDataWithVerification(
```
Comment (y0sher) [inline qbft/round_change.go:279]: I don't want to nit pick on the naming, but `WithVerification` and `IgnoreSignature` don't sound related to each other when and fact the functionality should be identical with one difference. 
I would go with either "WithVerification" and "NoVerification"
or "IgnoreSignature" and "WithSignature"/"VerifySignature"
but leaving it like now is a bit confusing imo

File: ssv/aggregator.go [hunk 22]
```diff
@@ -14,10 +14,11 @@ import (
 type AggregatorRunner struct {
 	BaseRunner *BaseRunner
 
-	beacon   BeaconNode
-	network  Network
-	signer   types.KeyManager
-	valCheck qbft.ProposedValueCheckF
+	beacon         BeaconNode
+	network        Network
+	signer         types.KeyManager
```
Comment (y0sher) [inline ssv/aggregator.go:19]: if I understand correctly at some other places you call this `types.SSVShareSigner`, it's best to decide on similar name/type. unless you create local interfaces for each struct with only the subset of functions they need.

File: ssv/validator.go [hunk 23]
```diff
@@ -10,26 +10,32 @@ import (
...
-	DutyRunners DutyRunners
-	Network     Network
-	Beacon      BeaconNode
-	Share       *types.Share
-	Signer      types.KeyManager
+	DutyRunners       DutyRunners
+	Network           Network
+	Beacon            BeaconNode
+	Share             *types.Share
+	Signer            types.KeyManager
```
Comment (y0sher) [inline ssv/validator.go:17]: ditto about SSVShare signer and keymanager ( and many other places )

File: qbft/round_change.go (hunk 21 above)
Comment (MatheusFranco99) [reply qbft/round_change.go:279]: Definitely. Forgot about that. Thanks!

File: ssv/aggregator.go (hunk 22 above)
Comment (MatheusFranco99) [reply ssv/aggregator.go:19]: The `KeyManager` includes the `SSVShareSigner`. Do you mean renaming the attribute `signer`?

File: types/signer.go [hunk 24]
```diff
@@ -32,13 +34,22 @@ type BeaconSigner interface {
...
 
-// SSVSigner used for all SSV specific signing
-type SSVSigner interface {
+// SSVOperatorSigner used for to sign protocol messages
+type SSVOperatorSigner interface {
+	SignSSVMessage(data []byte, pk []byte) ([]byte, error)
+	// AddSSVOperatorKey saves an SSV operator key
+	AddSSVOperatorKey(sk *rsa.PrivateKey) error
+	// RemoveSSVOperatorKey removes an SSV operator key
+	RemoveSSVOperatorKey(pubKey string)
```
Comment (y0sher) [inline types/signer.go:43]: IMO these are redundant, don't make us satisfy this interface (😆); There is always just a single operator key per instance, so there's never a need to add and remove them.

File: types/signer.go [hunk 25]
```diff
@@ -48,7 +59,7 @@ type DKGSigner interface {
 // KeyManager is an interface responsible for all key manager functions
 type KeyManager interface {
 	BeaconSigner
-	SSVSigner
+	SSVShareSigner
 	// AddShare saves a share key
 	AddShare(shareKey *bls.SecretKey) error
 	// RemoveShare removes a share key
```
Comment (y0sher) [inline types/signer.go:65]: mixing all this up is wrong if we're looking from SWE point.

**Review by y0sher — CHANGES_REQUESTED** [review]
some small stuff.

File: types/signer.go (hunk 25 above)
Comment (y0sher) [reply types/signer.go:65]: oh I see now it does not include the operator signer. sorry.

File: ssv/aggregator.go (hunk 22 above)
Comment (y0sher) [reply ssv/aggregator.go:19]: Oh I just saw SSVShareSigner is part of keymanager. sorry. I thought you splitted them somehow. resolving.

File: ssv/validator.go [hunk 26]
```diff
@@ -43,29 +49,63 @@ func (v *Validator) StartDuty(duty *types.Duty) error {
...
+	// Decode the nested SSVMessage
+	msg := &types.SSVMessage{}
+	if err := msg.Decode(signedSSVMessage.Data); err != nil {
+		return errors.Wrap(err, "could not decode data into an SSVMessage")
+	}
+
+	// Verify SignedSSVMessage's signature
+	if err := v.SignatureVerifier.Verify(signedSSVMessage, v.Share.Committee); err != nil {
+		return errors.Wrap(err, "SignedSSVMessage has an invalid signature")
+	}
```
Comment (moshe-blox) [inline ssv/validator.go:68]: we should probably decode after verifying

File: types/messages.go [hunk 27]
```diff
@@ -191,7 +191,7 @@ func (msg *SignedSSVMessage) Decode(data []byte) error {
 // - Data length should not be 0
 func (msg *SignedSSVMessage) Validate() error {
 	if msg.OperatorID == 0 {
-		return errors.New("OperatorID in SignedSSVMessage is 0")
+		return errors.New("signer ID 0 not allowed")
 	}
 	if len(msg.Signature) == 0 {
```
Comment (moshe-blox) [inline types/messages.go:196]: is there a reason to not require the expected length of RSA signatures, or is that not the right place to do so, or not the right assumption for the spec to make?

Comment (MatheusFranco99) [reply types/messages.go:196]: Actually, in the beginning, I wanted to let it open. But I've also added the custom decoding function to be used by implementation that sets it to 256 bytes
```go
signatureSize = 256
signature := encoded[signatureOffset : signatureOffset+signatureSize]
```
I have no opinion right now on whether we should check it here or leave the current check. 🤔

File: ssv/validator.go (hunk 26 above)
Comment (MatheusFranco99) [reply ssv/validator.go:68]: Agree. Thanks!

File: types/messages.go (hunk 27 above)
Comment (moshe-blox) [reply types/messages.go:196]: @GalRogozinski any reason why we shouldn't check that `len(sig) == signatureSize` in spec?

Comment (GalRogozinski) [reply types/messages.go:196]: When you decode the data from pubsub you create a byte array of the expected length
maybe change `SignedMessage` to have `Signature [256]byte`

btw @MatheusFranco99 I remember we said the encoded length of the sig we use is actually greater correct?

Comment (MatheusFranco99) [reply types/messages.go:196]: Ok, I'll change it to 256 bytes
This was about an encrypted key. For the signature, it's indeed 256

File: p2p/validation/msg_validation.go [hunk 28]
```diff
@@ -14,28 +14,13 @@ import (
...
 
-// To-Do: better spec the intended Message Validation behavior. Issue #375: https://github.com/bloxapp/ssv-spec/issues/375
-type MessageValidator interface {
-	ValidateSignedSSVMessage(runner ssv.Runner, signedSSVMessage *types.SignedSSVMessage) error
-	// Verifies the message's RSA signature with PKCSv1.5 encoding. Ref: https://datatracker.ietf.org/doc/html/rfc8017#section-8.2.2
-	VerifySignature(runner ssv.Runner, signedSSVMessage *types.SignedSSVMessage) error
-}
-
-func MsgValidation(runner ssv.Runner, msgValidator MessageValidator) MsgValidatorFunc {
+func MsgValidation(runner ssv.Runner) MsgValidatorFunc {
```
Comment (GalRogozinski) [inline p2p/validation/msg_validation.go:17]: do you see value to keep this file here for now?
if no, you may delete it

File: ssv/spectest/tests/runner/preconsensus/inconsistent_network_signer.go [hunk 29]
```diff
@@ -0,0 +1,72 @@
...
+import (
+	"github.com/attestantio/go-eth2-client/spec"
+
+	"github.com/bloxapp/ssv-spec/ssv/spectest/tests"
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/bloxapp/ssv-spec/types/testingutils"
+)
+
+// InconsistentNetworkSigner tests a SignedSSVMessage with inconsistent signer regarding its nested SignedPartialSignatureMessage
+func InconsistentNetworkSigner() tests.SpecTest {
```
Comment (GalRogozinski) [inline ssv/spectest/tests/runner/preconsensus/inconsistent_network_signer.go:12]: change network to operator on all tests I think

**Review by GalRogozinski — COMMENTED** [review]
just final nits and that's it

File: p2p/validation/msg_validation.go (hunk 28 above)
Comment (GalRogozinski) [reply p2p/validation/msg_validation.go:17]: doesn't have to be in this PR and not a blocker

**Review by moshe-blox — APPROVED** [review]

File: types/signer.go [hunk 30]
```diff
@@ -48,7 +54,7 @@ type DKGSigner interface {
 // KeyManager is an interface responsible for all key manager functions
 type KeyManager interface {
 	BeaconSigner
-	SSVSigner
+	ShareSigner
 	// AddShare saves a share key
 	AddShare(shareKey *bls.SecretKey) error
 	// RemoveShare removes a share key
```
[the remark this answers is not in the document]
Comment (MatheusFranco99) [reply types/signer.go:60]: Yes, other PR

File: p2p/validation/msg_validation.go (hunk 28 above)
Comment (MatheusFranco99) [reply p2p/validation/msg_validation.go:17]: Future PR

**Comment by y0sher** [comment]
Q: are we signing rsa in consensus ? If we are verifying the we should probably also sign.

**Comment by GalRogozinski** [comment]
> Q: are we signing rsa in consensus ? If we are verifying the we should probably also sign.
In spec we sign in validator.go

In impl from my POV you should sign where you pass all spec tests

**Review by GalRogozinski — APPROVED** [review]
